Tests/AsyncQueueTests/AsyncQueueTests.swift

```diff
@@ -26,8 +26,202 @@ import XCTest
 
 final class AsyncQueueTests: XCTestCase {
 
-    func test_example() {
-        _ = AsyncQueue()
+    // MARK: XCTestCase
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        systemUnderTest = AsyncQueue()
+    }
+
+    // MARK: Behavior Tests
+
+    func test_async_sendsEventsInOrder() async {
+        let counter = Counter()
+        for iteration in 1...1_000 {
+            systemUnderTest.async {
+                await counter.incrementAndExpectCount(equals: iteration)
+            }
+        }
+        await systemUnderTest.await { /* Drain the queue */ }
+    }
+
+    func test_async_executesAsyncBlocksAtomically() async {
+        let semaphore = Semaphore()
+        for _ in 1...1_000 {
+            systemUnderTest.async {
+                let isWaiting = await semaphore.isWaiting
+                // This test will fail occasionally if we aren't executing atomically.
+                // You can prove this to yourself by replacing `systemUnderTest.async` above with `Task`.
+                XCTAssertFalse(isWaiting)
+                // Signal the semaphore before or after we wait – let the scheduler decide.
+                Task {
+                    await semaphore.signal()
+                }
+                // Wait for the concurrent task to complete.
+                await semaphore.wait()
+            }
+        }
+        await systemUnderTest.await { /* Drain the queue */ }
+    }
+
+    func test_async_isNotReentrant() async {
+        let counter = Counter()
+        await systemUnderTest.await { [systemUnderTest] in
+            systemUnderTest.async {
+                await counter.incrementAndExpectCount(equals: 2)
+            }
+            await counter.incrementAndExpectCount(equals: 1)
+            systemUnderTest.async {
+                await counter.incrementAndExpectCount(equals: 3)
+            }
+        }
+        await systemUnderTest.await { /* Drain the queue */ }
+    }
+
+    func test_async_retainsReceiverUntilFlushed() async {
```

Comment:
What is the receiver in this context?

Reply:
The receiver is the `systemUnderTest`, i.e. the async queue.

Reply:
Ah got it. I understand now 👍